pkg/cloud/network.go

```diff
@@ -205,7 +175,19 @@ func (c *client) AssociatePublicIPAddress(csCluster *infrav1.CloudStackCluster)
 	if _, err := c.cs.Address.AssociateIpAddress(p); err != nil {
 		return err
 	}
-	return nil
+	return c.AddClusterTag(ResourceTypeIPAddress, publicAddress.Id, csCluster, !alreadyAllocated)
+}
+
+func (c *client) DisassociatePublicIPAddress(csCluster *infrav1.CloudStackCluster) (retErr error) {
```

Comment:
Please add comments to the top of all new exported functions.

Wonkun's PR brings in linting for this, so it will soon be required.

Make sure to start the comment with the function name and place it right above the function. Most IDEs pick it up this way.

Reply:
Done.  Thanks!